octoprint_bambu_printer/printer/file_system/ftps_client.py

```diff
@@ -185,19 +144,26 @@ def move_file(self, source: str, dest: str):
     def mkdir(self, path: str) -> str:
         return self.ftps_session.mkd(path)
 
-    def list_files(self, path: str, file_pattern: Optional[str] = None) -> Union[List[str], None]:
+    def list_files(
+        self, list_path: str, extensions: str | list[str] | None = None
+    ) -> Generator[Path]:
         """list files under a path inside the FTPS server"""
+
+        if extensions is None:
+            _extension_acceptable = lambda p: True
+        else:
+            if isinstance(extensions, str):
+                extensions = [extensions]
+            _extension_acceptable = lambda p: any(s in p.suffixes for s in extensions)
+
         try:
-            files = self.ftps_session.nlst(path)
-            if not files:
-                return
-            if file_pattern:
-                return [f for f in files if file_pattern in f]
-            return files
+            list_result = self.ftps_session.nlst(list_path) or []
+            for file_name in list_result:
+                path = Path(list_path) / file_name
+                if _extension_acceptable(path):
+                    yield path
```

Comment:
I think this is where the issue is with the timelapse folder listing stuff and the duplicated pathing.

Reply:
In the original code I had some stupid logic to get around this. I can't remember if it's because of a difference in how X1 list files on ftps vs A1/P1 series printers.

https://github.com/jneilliii/OctoPrint-BambuPrinter/blob/ad862d5ebd4b0579d52923c2725ea4c7a7cc137e/octoprint_bambu_printer/__init__.py#L154-L158

Reply:
The equivalent of this operation would be to use Path(filename).name this accounts for most operations you've done

Reply:
I see the issue with that filename. I use parent folder as base and append a filename to it. but filename already includes folder name, since listing files for p1s gives only names, but in your case maybe it lists them with folder name

Reply:
Exactly this, the different printer models give the list in different ways. Maybe the more generic solution to my hack is replacing list_path in filename...

```
path = Path(list_path) / file_name.replace(list_path, "")
```

Reply:
Fixed this issue, covered with tests and improved logging for ftp connection.

Reply:
yep, works perfect now with timelapse listing and loading thumbnails. running through some real world tests now with a print.